### PR TITLE
Replace libhttpsimple with direct libcurl binding

### PR DESCRIPTION
**Pull request by mitkus**
Makes project easier to build, no more C library compilation necessary.

I've tested in on my Mac OS 10.3 and successfully built ubuntu-16-04 target. Probably good idea to try out a few more machines before releasing, considering all native library issues that we've had already.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,4 +19,4 @@ before_install:
   - bash setup-travis.sh
 script:
   - sbt clean test package buildAll
-  - LD_LIBRARY_PATH=libhttpsimple/target cli/target/scala-2.11/reactive-cli-out --help
+  - LD_LIBRARY_PATH=cli/target/scala-2.11/reactive-cli-out --help
```
Comment (longshorej) [inline .travis.yml]: LD_LIBRARY_PATH= should be removed, i.e. line should be just `cli/target/scala-2.11/reactive-cli-out --help`

File: libhttpsimple-bindings/src/main/scala/libhttpsimple/LibHttpSimple.scala [hunk 2]
```diff
@@ -158,12 +158,11 @@ object LibHttpSimple {
       result
     }
 
-  private def httpHeadersToDelimitedString(headers: Map[String, String]): String =
+  private def httpHeadersToDelimitedString(headers: Map[String, String]): Seq[String] =
```
Comment (longshorej) [inline libhttpsimple-bindings/src/main/scala/libhttpsimple/LibHttpSimple.scala]: A bit of a Scala wart here - could you add `import scala.collection.immutable.Seq` at the top of the file? The default `Seq` isn't immutable (to support varg syntax with arrays), but will be in a future release AFAIK.

File: libhttpsimple-bindings/src/main/scala/libhttpsimple/LibHttpSimple.scala [hunk 3]
```diff
@@ -158,12 +158,11 @@ object LibHttpSimple {
...
     headers
       .map {
         case (headerName, headerValue) => s"$headerName$HttpHeaderNameAndValueSeparator $headerValue"
-      }
-      .mkString(CRLF)
+      }.toList
```
Comment (longshorej) [inline libhttpsimple-bindings/src/main/scala/libhttpsimple/LibHttpSimple.scala]: Just an aside -- Vectors perform better in the typical case. List works well when you don't need to know the length and are building it up via prepending elements, usually in simple tail recursive functions. Either works here, just thought I'd mention.

**Comment by fsat** [comment]
@mitkus - nicely done! Thanks for getting around to doing this!

**Comment by longshorej** [comment]
Build failed - seems RPM related.

**Comment by longshorej** [comment]
I added a few commits to this PR to reduce the dependency on random sites e.g. llvm.org and speed up build times.

1) Implemented procedure for building images and publishing them to Bintray.
2) Updated builds to use these published images
3) Removed custom argonaut building as 6.2.1 is officially published now, with native support.
4) Added cache for docker images to the travis build
5) Removed building of re2 and other deps for travis, as it's redundant given travis runs `buildAll`.

I'm happy with Travis build times now. There exists the possibility of further reducing them by only building a single rpm/deb. If that becomes necessary in the future, then we should setup nightly builds to build everything.

**Pull request merged**